### ISSUE TITLE
Feature Request: support «writing-mode:sideways-lr»

### ISSUE THREAD

**Issue by stolzweb** [Kozea/WeasyPrint#577]
We tried nearly any html to pdf converter. But non of them could reach your level. Amazing work! As we need to rotate text in a table 90 deg we tried to use transform: rotate(90deg);. But CSS does not exatly what we expected. The css tag «writing-mode:sideways-lr» does exactly what we expected, but that's not supported in WeasyPrint. Is there any chance that WeasyPrint can support «writing-mode:sideways-lr»?

**Comment by liZe** [comment]
> We tried nearly any html to pdf converter. But non of them could reach your level. Amazing work!

Thank you!

> Is there any chance that WeasyPrint can support «writing-mode:sideways-lr»?

I'm currently working on right-to-left and bidirectional text, it's a first step before `writing-mode`. It's really not easy to implement, so I can't promise a deadline, but it would be a great feature to have!

**Comment by shadiakiki1986** [comment]
Hey there @liZe 
Any news on right-to-left text?

**Comment by liZe** [comment]
> Any news on right-to-left text?

Hi!

There's no new code. You can follow #106 that's on [my short TODO-list before 43](https://github.com/Kozea/WeasyPrint/milestone/10).

**Comment by sravankml** [comment]
even   writing-mode: vertical-rlis not supporting

**Comment by claell** [comment]
I just ran into this. Is there any update?

For now, a workaround for vertical text would be very appreciated.

`transform: rotate(270deg);` seems to be something, but breaks a lot of orientation stuff.

**Comment by liZe** [comment]
> I just ran into this. Is there any update?

There’s no update about this, it’s a huge feature that requires a lot of changes.

**Comment by danielricecodes** [comment]
```bash
% weasyprint --version
WeasyPrint version 65.1
```

Still no `writing-mode` support in August 2025.  This is a bit frustrating since vertical column headers are not uncommon.  

However, I was able to work around the problem with `transform: rotate(-90deg);` and applying padding to get the content into the right location.  I'm still trying to massage the table column widths, but I get this when I wrap the table cell contents in a `div`.

```css
.side-effects-table tr:first-child td:not(:first-child) div { 
  transform: rotate(-90deg);
  padding-left: 8mm;
  padding-bottom: 10mm;
  margin: 0;
}
```

And it looks like this, which is better than nothing:

<img width="910" height="555" alt="Image" src="https://github.com/user-attachments/assets/97da1ec0-1ee9-4165-a546-b7bd1c69a11e" />

**Comment by liZe** [comment]
> Still no `writing-mode` support in August 2025. This is a bit frustrating since vertical column headers are not uncommon.

Yes, this is a bit frustrating. Don’t hesitate to open a pull request to fix the problem, or to sponsor the development of the feature, so that the frustration can go away!

**Comment by danielricecodes** [comment]
> > Still no `writing-mode` support in August 2025. This is a bit frustrating since vertical column headers are not uncommon.
> 
> Yes, this is a bit frustrating. Don’t hesitate to open a pull request to fix the problem, or to sponsor the development of the feature, so that the frustration can go away!

I appreciate the support but I'm a Ruby developer mostly. I would try but I'm a bit out of my element in this project I'm afraid 😱
